dev/devicelab/lib/framework/devices.dart

```diff
@@ -848,6 +861,23 @@ class AndroidDevice extends Device {
   Future<void> reboot() {
     return adb(<String>['reboot']);
   }
+
+  @override
+  Future<void> awaitDevice() async {
+    print('Waiting for device.');
+    final String waitOut = await adb(<String>['wait-for-device']);
+    print(waitOut);
+    const RetryOptions retryOptions = RetryOptions(delayFactor: Duration(seconds: 1), maxAttempts: 30, maxDelay: Duration(minutes: 1));
```

Comment:
Can this be translated to a one liner like https://cs.opensource.google/flutter/recipes/+/master:recipe_modules/android_virtual_device/resources/avd_setup.sh;l=15?

Reply:
I tried that and the android shell would not run it. This is also a safer method as this will not potentially loop forever.